sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_floating_point.asciidoc

```diff
@@ -0,0 +1,1160 @@
+OpArbitraryFloatATanPiINTEL
+OpArbitraryFloatATan2INTEL
+OpArbitraryFloatPowINTEL
+OpArbitraryFloatPowRINTEL
+OpArbitraryFloatPowNINTEL
+OpArbitraryFloatCastINTEL
+OpArbitraryFloatCastFromIntINTEL
+OpArbitraryFloatCastToIntINTEL
+----
+
+== Token Number Assignments
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|`ArbitraryPrecisionFloatingPointINTEL`   | 5845
+|`OpArbitraryFloatAddINTEL`               | 5846
+|`OpArbitraryFloatSubINTEL`               | 5847
+|`OpArbitraryFloatMulINTEL`               | 5848
+|`OpArbitraryFloatDivINTEL`               | 5849
+|`OpArbitraryFloatGTINTEL`                | 5850
+|`OpArbitraryFloatGEINTEL`                | 5851
+|`OpArbitraryFloatLTINTEL`                | 5852
+|`OpArbitraryFloatLEINTEL`                | 5853
+|`OpArbitraryFloatEQINTEL`                | 5854
+|`OpArbitraryFloatRecipINTEL`             | 5855
+|`OpArbitraryFloatRSqrtINTEL`             | 5856
+|`OpArbitraryFloatCbrtINTEL`              | 5857
+|`OpArbitraryFloatHypotINTEL`             | 5858
+|`OpArbitraryFloatSqrtINTEL`              | 5859
+|`OpArbitraryFloatLogINTEL`               | 5860
+|`OpArbitraryFloatLog2INTEL`              | 5861
+|`OpArbitraryFloatLog10INTEL`             | 5862
+|`OpArbitraryFloatLog1pINTEL`             | 5863
+|`OpArbitraryFloatExpINTEL`               | 5864
+|`OpArbitraryFloatExp2INTEL`              | 5865
+|`OpArbitraryFloatExp10INTEL`             | 5866
+|`OpArbitraryFloatExpm1INTEL`             | 5867
+|`OpArbitraryFloatSinINTEL`               | 5868
+|`OpArbitraryFloatCosINTEL`               | 5869
+|`OpArbitraryFloatSinCosINTEL`            | 5870
+|`OpArbitraryFloatSinPiINTEL`             | 5871
+|`OpArbitraryFloatCosPiINTEL`             | 5872
+|`OpArbitraryFloatSinCosPiINTEL`          | 5840
+|`OpArbitraryFloatASinINTEL`              | 5873
+|`OpArbitraryFloatASinPiINTEL`            | 5874
+|`OpArbitraryFloatACosINTEL`              | 5875
+|`OpArbitraryFloatACosPiINTEL`            | 5876
+|`OpArbitraryFloatATanINTEL`              | 5877
+|`OpArbitraryFloatATanPiINTEL`            | 5878
+|`OpArbitraryFloatATan2INTEL`             | 5879
+|`OpArbitraryFloatPowINTEL`               | 5880
+|`OpArbitraryFloatPowRINTEL`              | 5881
+|`OpArbitraryFloatPowNINTEL`              | 5882
+|`OpArbitraryFloatCastINTEL`              | 5841
+|`OpArbitraryFloatCastFromIntINTEL`       | 5842
+|`OpArbitraryFloatCastToIntINTEL`         | 5843
+|====
+
+== Modifications to the SPIR-V Specification Version 1.5
+
+After Section 3.16, add a new section "3.16a Subnormal Support" as follows:
+
+=== Subnormal Support
+
+Control whether subnormal support is enabled or not.
+
+[options="header"]
+[width="60%"]
+[cols="15%,85%"]
+|====
+| Value ^| Subnormal Support (`OpTypeInt`)
+| 0 | Flush subnormals numbers to zero on inputs and outputs
+| 1 | Enable support for operating on subnormal numbers
+|====
+
+After Section 3.16, add a new section "3.16d Rounding Accuracy" as follows:
+
+=== Rounding Accuracy
```

Comment:
If I understand correctly, all instructions documented here have both "Subnormal Support" and "Rounding Accuracy" as literal arguments - this is a 8 bytes for storing 4 bits, which is kind of overkill.

This is not a major comment, but probably you will consider merging these two into a single bitmask and reducing amount of arguments of each new instructions.

The same applies to fixed-point extension

Reply:
This one's on me - we could change this, but since the translator is already using the preliminary spec, we'll have to circle back and update it there as well. Can we table this one for a later changelist? Also, do internal SPIR-V API changes need a deprecation period as well?